### ISSUE TITLE
Centered editor layout and toggling the side bar should preferably not resize the sashes

### ISSUE THREAD

**Issue by thernstig** [Microsoft/vscode#60900]
With centered editor layout enabled (with or without zen mode), toggling the side bar on/off will resize the sashes. Even though there is enough real-estate for the side bar to show without doing so. To illustrate this behavior, look at the right sash in both these screenshots.

1.
![1](https://user-images.githubusercontent.com/30827238/46919994-db64a000-cfe7-11e8-8ff8-7d718cf85880.PNG)

2.
![2](https://user-images.githubusercontent.com/30827238/46919995-de5f9080-cfe7-11e8-98ae-44910a3764db.PNG)

This illustrates the "jumpiness" when opening the side bar in this scenario. It would be preferable that the sashes were not moved unless needed.

**Comment by thernstig** [comment]
@isidorn Ah yes you are correct. For me it makes more sense to center at workbench area then editor area, but maybe that is just me. It would feel more natural, especially when opening/hiding parts on the sides. If there is free room, the sashes would not move at all which would be pleasant I think.

**Comment by isidorn** [comment]
@Sourgin you should be possible to change the sash border by draging. Also alt + drag only one side also works.
If it does not pleas file a new issue with reproducable steps

![drag](https://user-images.githubusercontent.com/1926584/47646440-34017480-db74-11e8-9316-278f816d0180.gif)

**Comment by zhiayang** [comment]
Is this feature on the roadmap at all? Sure, I can use alt+drag to resize the left edge so I get the editor view centred, but the moment I close the sidebar, the editor is now too far to the left, so I need to adjust the view again.

I'm not sure if this is difficult to implement, but this behaviour was (inexplicably) changed about a year ago, and basically nobody has acknowledged that this is an issue, and from what I can gather you can't make an extension to change this behaviour.

**Comment by thernstig** [comment]
@Sourgin @isidorn Please update the issue so it does not get closed :) I think this should be a simple change and I think it's a legit concern that makes the centered editor layout less than ideal (I stopped using it) due to this reason. It is just too jumpy when opening/closing the side-bar. The major problem is that often I leave about 120 char width for both left/right side in a split editor mode and when I open the sidebar text might only get <100 char width which hides code.

**Comment by Yajo** [comment]
Actually the problem is even worse. Basically, centered layout is completely useless as it is right now. Let me explain.

Problem 1 and 2: the editor is centered in the editor area, not the workbench (this is what is explained in https://github.com/microsoft/vscode/issues/60900#issue-369924671). So, if I toggle any side panel, the editor shows 2 problems: it is *not centered* and it *shrinks*. None of these should happen. If it is *not centered*, it can hurt your neck if you use it for a long time. If it *shrinks*, your code may be hidden/wrapped and you're incomfortable. See, I configured here a text editor of about 130 chars width. Look at line 51:

![Peek 17-01-2020 09-45](https://user-images.githubusercontent.com/973709/72597465-24ec1b00-390e-11ea-9d24-45ce50d4e133.gif)

Solution:

- vscode shouldn't store the % of margin I'm choosing, but the editor width I'm choosing.
- when the area goes below the chosen width, just let the editor use 100% of its area.
- center the editor in the window, not the area.
- if there's a panel in the left or right side whose width is bigger than the margin needed to center the editor, just respect the editor width and slide it a little bit to the other side as long as there's room for it.

Problem 3: the centered layout setting is stored per project, which doesn't make any sense. Do you use different necks for each project? It's like storing the editor theme per project... Example: you change to dark theme when it's dark, and to light theme during the day; depends on ambient. Here also: you center layout when your screen is too big, and you may disable it when it's narrower (although if applied the fixes above, you could just enable centered always). This is a view setting, and it should be globally enabled until globally disabled, no more. Look, when I switch projects:

![Peek 17-01-2020 09-50](https://user-images.githubusercontent.com/973709/72597848-e2770e00-390e-11ea-93ab-f0a79f6ab196.gif)

Solution:

- treat this setting as a view option, just like the theme.

**Comment by thernstig** [comment]
@yajo Yes, your problem 1 and 2 is what this is about. Regarding problem 3, I am not sure what you mean. Just save the setting as user settings instead of work space settings.

**Comment by zhiayang** [comment]
exactly, i'm glad i'm not the only one that can't use centred layout because of this. With a monitor larger than say 24", it really starts to get noticeable that I have to move my neck every time i open/close the sidebar.

**Comment by Yajo** [comment]
> Just save the setting as user settings instead of work space settings

That's not possible, the setting is not stored anywhere, not in user nor workspace settings. It's somehow hidden.

**Comment by Yajo** [comment]
> Just save the setting as user settings instead of work space settings

That's not possible, the setting is not stored anywhere, not in user nor workspace settings. Vscode saves it somewhere hidden, but per workspace always and that cannot be changed.

If all other problems are fixed, adding the setting to user settings.json file would be a good fix for problem 3.

**Comment by thernstig** [comment]
@Yajo @zhiayang @Sourgin @kutyei @pedrobaeza @yenthe666 @alexhafner @nufflee et. al. since they have such a tough requirement of 20! upvotes I request you all bring some friends in to upvote this if they agree :)

**Comment by thernstig** [comment]
@isidorn Isn't 20 upvotes a little harsh for this, as the feature request makes perfect sense?

**Comment by isidorn** [comment]
I will move it to the backlog.

**Comment by Yajo** [comment]
We are 20 now :sunglasses:

**Comment by vishwa-raj** [comment]
Any update on this issue ?
It hasn't even moved to [On Deck](https://github.com/microsoft/vscode/milestone/27) yet

**Comment by vistuleB** [comment]
Same issue.

It would be great if the sash positions were recorded as absolutes with respect to the the workbench window. (At least until the workbench window is itself resized.)

That way, when I open the terminal panel on the right on a small (laptop) screen the right-hand sash could/should get completely covered over by the terminal panel instead of causing my centered layout to jump left, with the right-hand sash suddenly wasting now-valuable real estate.

**Comment by artus9033** [comment]
The behaviour still hasn't changed and is very irritating when enabling the centered layout mode with the left sidebar (which is a common setup, since the file navigator is located there). Is there any chance this feature will ever even be placed in On Deck?

**Comment by FancyBanana** [comment]
@thernstig Replying to your message, while the issue in the same category, this one is more complex. From what I saw editor width is controlled by the editor area, making it centered would require knowing the sized of all side elements, which would require passing that information through the common ancestor of all those elements or creating a separate component for tracking that information. If @sbatten  can point me to the right location I will attempt this since I would also prefer the editor to not to jump, but no promises.

**Comment by lemoscaio** [comment]
I would love this feature, too. I use a 49" super ultrawide monitor, and I'd like to leave VSCode maximized but with the editor centered on my screen. Not having it centered when I open the sidebar is frustrating because of the monitor size.

Centering it to the application viewport instead of the editor's available space would be great, and this goes well with another feature I would love: floating sidebars that show over the editor, not squeezing everything.

**Comment by Sourgin** [comment]
Hi, it has been 7 years since the original comment, I have received a couple of e-mails each year about this, I keep reading you all.

Since the original post, almost a decade old now we had new technology emerge completely, like Copilot, might be wise to give it a try.

**Comment by Wim-De-Clercq** [comment]
Is it 
```
  "workbench.editor.centeredLayoutFixedWidth": true,
```
you all are looking for, or am I misunderstanding the problem?

**Comment by p-dros** [comment]
Is there any work planned for this issue?

**Comment by Sourgin** [comment]
Captain log, august 3 2026.
Engine room keeps ignoring our messages.
The newly employes that replaced the old ones are as usefull as previous ones.
They don't know how to simply resize a layout based on window size and not arbitrary work area designated by them.

Captain out.
